### ISSUE TITLE
What is the future of the project?

### ISSUE THREAD

**Issue by vedmalex** [apollostack/saturn#31]
There is need to move to current apollo-server, because is it support not only express but also `hapi`, `koa`, and `connect`!
also we need to move it into next apollo-client because currently there is issue with ssr of live queries? which is solver in newer version.  

May be [satrun](https://github.com/apollostack/saturn/) must be something like [Create React App](https://github.com/facebookincubator/create-react-app) but for apollo-stack. Why not?

Even though [Why frameworks are evil](http://tomasp.net/blog/2015/library-frameworks/) i've use and build one for myself :)

So, I think Issue #30 is just for start.

- [ ] moving to next version of apollo-server
- [ ] make configuration more flexiable on the derived app
- [ ] koa template
- [ ] connect template
- [ ] hapi template
- [ ] move to TypeScript, to make it more mature. (Why not? i think)

**Comment by tmeasday** [comment]
Hey @vedmalex good questions.

I'd like to revisit the rationale for Saturn in the context of `create-react-app`. I think there's no point in trying to replicate what CRA does (client side build / dev server). So ideally we should work towards removing that part of Saturn and focussing on supplying the SSR layer and the apollo API server.

Do you think there's a whole lot of value in maintaining separate templates for the different routing libraries? What is the use case you imagine we are targeting there? I would suggest if people want to add "server only" routes, they might consider standing up another microservice. Or do you think this is making things too complicated? Alternatively the `apiServer` file is pretty simple really, it's not that hard to analogize it yourself for a different routing framework.

I don't think we should migrate to typescript. The rationale is that we want to offer the "escape hatch" of "copy code out of saturn into your app" if you need to extend things past where the framework allows. If the code is written in TS, this is a big barrier for a lot of people.

**Comment by HriBB** [comment]
> copy code out of saturn into your app
Exactly! Keep up the good work ;)

**Comment by tmeasday** [comment]
> Saturn could provide a simple SSR layer for apollostack

The tricky part is that you do need to stand up an app server (or an "SSR server") as well as an API server [1], which isn't something that `create-react-app` does out of the box (although I have seen some extensions of CRA that are starting to do things like that.)

I'm not sure what the best approach to the goal of "allow people to do SSR in Apollo as easily as possible" is right now, it's probably something we need to think about a little more soon. cc @stubailo 

[1] They could be the same server of course, it's debatable.

**Comment by vedmalex** [comment]
@tmeasday @HriBB , frankly, 
I'm not sure about typescript, but apollo-server/-client now on typescript, isn't it?
yet another is angular2.

there a lot of bug yet(in typescript)... i've using it full time for a five days :) a lot of fun... visual studio code has fulfeatured debug tool it is great! 
i'm not follower of the java backended IDE for javascript like storm, eclipse..., code(-insiders)/atom/sublime-text/vim... just right tool for specific thing.

why typescript may be the solution? b/c of it `strict type` javascript, but there is also `flow`.

I think that framework must be flexible and preatty simple to use, as it is now.

**Comment by vedmalex** [comment]
> I'd like to revisit the rationale for Saturn in the context of create-react-app. I think there's no point in trying to replicate what CRA does (client side build / dev server). So ideally we should work towards removing that part of Saturn and focussing on supplying the SSR layer and the apollo API server. Perhaps we can embed CRA's react-tools as a dependency.

this is exaclty what I think about now! even more we can emded a relay support here also for SSR and api. what you think @tmeasday ?

**Comment by vedmalex** [comment]
>Do you think there's a whole lot of value in maintaining separate templates for the different routing libraries?


not at all! but just see apollo and findout that may be one will wants to :). i'm using express from node v4 and still happy with it ! but may one will want to contribute?
 
>What is the use case you imagine we are targeting there? I would suggest if people want to add "server only" routes, they might consider standing up another microservice. Or do you think this is making things too complicated?

microservices is good approach also, but some times I want to run it from one place. to debug every thing in one place and not to have mix the debug port and so on... i'm not writing the test all the time :) some thing need to be discovered, and there is no way to debug it with any test case.

>Alternatively the apiServer file is pretty simple really, it's not that hard to analogize it yourself for a different routing framework.

very simple indeed!

**Comment by jperl** [comment]
A `create-react-app` for Typescript + React + Apollo + Hot Loader 3 would be amazing.

**Comment by jperl** [comment]
There is definitely a huge pain in configuring webpack -- compared to meteor's build tool which was very simple to add build plugins to.

I could imagine a meteor build tool v2 -- which is like `create-react-app`. Built on top of webpack, very easy to add build plugins to (typescript, tslint, etc), and there is an escape hatch back to pure npm.

**Comment by tmeasday** [comment]
@jperl I tend to agree with you. This is probably the closest thing to that right now; maybe it can be a good base to get you there!

**Comment by jperl** [comment]
In case anyone is looking for Typescript + React + Hot Reload, checkout https://github.com/piotrwitek/react-redux-typescript-starter-kit.

After trying many things (webpack 2 + HL3, browserify + tsify + watchify, gulp) it is by far the fastest.
